File: src/main/java/no/nav/syfo/utils/JAXB.java
package no.nav.syfo.utils;


import no.nav.melding.virksomhet.sykepengesoeknadoppsummering.v1.sykepengesoeknadoppsummering.ObjectFactory;

import javax.xml.bind.JAXBContext;
import javax.xml.bind.JAXBException;
import java.io.StringReader;

import static javax.xml.bind.JAXBContext.newInstance;

public class JAXB {

    private static final JAXBContext SYKEPENGESOEKNAD_OPPSUMMERING_CONTEXT;

    static {
        try {
            SYKEPENGESOEKNAD_OPPSUMMERING_CONTEXT = newInstance(
                    ObjectFactory.class
            );
        } catch (JAXBException e) {
            throw new RuntimeException(e);
        }
    }

    @SuppressWarnings("unchecked")
    public static <T> T unmarshalSykepengesoknadOppsummering(String melding) {
        try {
            return (T) SYKEPENGESOEKNAD_OPPSUMMERING_CONTEXT.createUnmarshaller().unmarshal(new StringReader(melding));
        } catch (JAXBException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/no/nav/syfo/util/JAXB.kt
package no.nav.syfo.util

import no.nav.melding.virksomhet.sykepengesoeknadoppsummering.v1.sykepengesoeknadoppsummering.ObjectFactory
import java.io.StringReader
import javax.xml.bind.JAXBContext
import javax.xml.bind.JAXBException

object JAXB {
    private var SYKEPENGESOEKNAD_OPPSUMMERING_CONTEXT: JAXBContext
    @JvmStatic
    fun <T> unmarshalSykepengesoknadOppsummering(melding: String?): T {
        return try {
            SYKEPENGESOEKNAD_OPPSUMMERING_CONTEXT.createUnmarshaller().unmarshal(StringReader(melding)) as T
        } catch (e: JAXBException) {
            throw RuntimeException(e)
        }
    }

    init {
        SYKEPENGESOEKNAD_OPPSUMMERING_CONTEXT = try {
            JAXBContext.newInstance(
                ObjectFactory::class.java
            )
        } catch (e: JAXBException) {
            throw RuntimeException(e)
        }
    }
}
